### PR TITLE
Chore/document term description

### PR DESCRIPTION
**Pull request by pmcb55**
# New feature description
Term description meta-data now always displayed for all terms (regardless of  it being non-English or not), and more detailed descriptions provided too (and all with full test coverage).

# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] The changelog has been updated, if applicable.
- [X] New functions/types have been exported in `index.ts`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/DatasetHandler.test.js [hunk 1]
```diff
@@ -248,10 +303,46 @@ describe("Dataset Handler", () => {
...
+        inputResources: ["does not matter"],
+      });
+
+      const result = handler.buildTemplateInput();
+      expect(result.classes.length).toEqual(1);
+      const description = result.classes[0].termDescription;
+      expect(description).toContain("descriptions only in English");
+    });
+
+    it("should describe single lable and comment in no-locale only", () => {
```
Comment (andydavison) [inline src/DatasetHandler.test.js:329]:
```suggestion
    it("should describe single label and comment in no-locale only", () => {
```

File: test/resources/expectedOutputs/java-rdf4j/src/main/java/com/inrupt/testing/SCHEMA_INRUPT_EXT.java [hunk 2]
```diff
@@ -55,23 +55,23 @@ private static String FULL_IRI(final String localName) {
     /**
      * A person (alive, dead, undead, or fictional).
      *
-     * This term has [4] labels and comments, in the languages [de, es, fr, it].
+     * This term provides multilingual descriptions, but has a mismatch between its labels and comments, with [6] labels in languages [NoLocale, de, en, es, fr, it], but [5] comments in languages [NoLocale, de, es, fr, it].
```
Comment (andydavison) [inline test/resources/expectedOutputs/java-rdf4j/src/main/java/com/inrupt/testing/SCHEMA_INRUPT_EXT.java:58]: I'm assuming this sort of thing is a schema.org issue?

File: src/DatasetHandler.test.js (hunk 1 above)
Comment (pmcb55) [reply src/DatasetHandler.test.js:329]: Yep.

File: test/resources/expectedOutputs/java-rdf4j/src/main/java/com/inrupt/testing/SCHEMA_INRUPT_EXT.java (hunk 2 above)
Comment (pmcb55) [reply test/resources/expectedOutputs/java-rdf4j/src/main/java/com/inrupt/testing/SCHEMA_INRUPT_EXT.java]: It's good that you pointed this out (I was aware of it, but kinda brushed it aside (as I understand why it's happening) - but it's ripe for causing other people confusion too).
So no, this comes from our sample extension to Schema.org ('cos Schema.org doesn't provide any translations at all today) - so it's our simply extension that is providing these translations (i.e., test/resources/vocabs/schema-inrupt-ext.ttl).
But in fact, this does illustrate an interesting edge case (caused by the frustrating difference between RDF string literals that have explicit language tags, and those that don't (e.g. `<> rdfs:label "My label"@en .` and `<> rdfs:label "My label" .`)).
In this particular case, the original Schema.org does provide a comment for `schema:Person` in English, but they don't provide an explicit language tag (which I actually consider 'bad form' from them, but it's very common in vocabs out there in the wild) - here is the official Schema.org Turtle (viewable from [here](https://schema.org/docs/developers.html)):

```turtle
schema:Person a rdfs:Class ;
    rdfs:label "Person" ;
    rdfs:comment "A person (alive, dead, undead, or fictional)." ;
    rdfs:subClassOf schema:Thing ;
    ns3:equivalentClass <http://xmlns.com/foaf/0.1/Person> ;
    schema:source <http://www.w3.org/wiki/WebSchemas/SchemaDotOrgSources#source_rNews> .
```

So our AG will look for comments with either "@en", or no language tag at all (and just assume that it's in English!). Therefore our extension doesn't need to duplicate the `rdfs:comment` from the original Schema.org definition. Although it could do that, and then stick an explicit "@en" on it's copy, which would then remove this apparent 'mismatch' (and arguably would be more correct (i.e., since it wouldn't rely on people looking for "@en", and if not found then having to fall back to looking for no language tag and if found _assuming_ that that value is actually in English)).
Or, I could just detect a mismatch that is based purely on a "@en" vs "NoLocale" difference, and remove the "but has a mismatch" wording. I've implemented that now: e7a5c94

**Review by andydavison — APPROVED** [review]
approved - I agree that the soft assumption that 'no locale' is _probably_ English minimises confusion with those common mismatches
